nekoyume/Assets/_Scripts/UI/Scroller/RankCellPanel.cs

```diff
@@ -122,11 +97,68 @@ public void SetDataAsStage(StageRankingModel rankingInfo)
             firstElementText.text = rankingInfo.ClearedStageId.ToString();
             firstElementText.gameObject.SetActive(true);
             firstElementCpText.gameObject.SetActive(false);
-            secondElement.gameObject.SetActive(false);
+            secondElementText.gameObject.SetActive(false);
+            secondElementEquipmentNameText.gameObject.SetActive(false);
 
-            var rank = rankingInfo.Rank;
+            UpdateRank(rankingInfo.Rank);
+            characterView.SetByAvatarState(rankingInfo.AvatarState);
+            gameObject.SetActive(true);
+        }
+
+        public void SetDataAsCrafting(CraftRankingModel rankingInfo)
+        {
+            nicknameText.text = rankingInfo.AvatarState.name;
+            nicknameText.gameObject.SetActive(true);
+            addressText.text = rankingInfo.AvatarState.address
+                .ToString()
+                .Remove(addressStringCount);
+
+            firstElementText.text = rankingInfo.CraftCount.ToString();
+            firstElementText.gameObject.SetActive(true);
+            firstElementCpText.gameObject.SetActive(false);
+            secondElementText.gameObject.SetActive(false);
+            secondElementEquipmentNameText.gameObject.SetActive(false);
+
+            UpdateRank(rankingInfo.Rank);
+            characterView.SetByAvatarState(rankingInfo.AvatarState);
+            gameObject.SetActive(true);
+        }
+
+        public void SetDataAsEquipment(EquipmentRankingModel rankingInfo)
+        {
+            nicknameText.text = rankingInfo.AvatarState.name;
+            nicknameText.gameObject.SetActive(true);
+            addressText.text = rankingInfo.AvatarState.address
+                .ToString()
+                .Remove(addressStringCount);
+
+            firstElementCpText.text = rankingInfo.Cp.ToString();
+
+            var equipmentItemSheet = Game.Game.instance.TableSheets.EquipmentItemSheet;
+            secondElementEquipmentNameText.text = LocalizationExtension.GetLocalizedName(
+                equipmentItemSheet,
+                rankingInfo.EquipmentId,
+                rankingInfo.Level);
+
+            firstElementText.gameObject.SetActive(false);
+            firstElementCpText.gameObject.SetActive(true);
+            secondElementText.gameObject.SetActive(false);
+            secondElementEquipmentNameText.gameObject.SetActive(true);
+
+            UpdateRank(rankingInfo.Rank);
+            characterView.SetByAvatarState(rankingInfo.AvatarState);
+            gameObject.SetActive(true);
+        }
```

Comment:
I think you can make `SetDataAsEquipment()` and `SetDataAsCrafting()` into one using generics.